### PR TITLE
Add full para constructor

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Fixes https://github.com/Azure/azure-sdk-for-go/issues/12834

### CODE REVIEW

File: src/Templates/ModelTemplate.cshtml [hunk 1]
```diff
@@ -414,8 +414,11 @@ if (modelType is DictionaryTypeGo dictionaryType && dictionaryType.SupportsAddit
         }
 
         // Creates a new instance of the @Model.Name type.
-        func New@(Model.Name) (getNextPage @(modelPageType.NextPageFunctionSig)) @Model.Name {
-            return @(Model.Name){@(modelPageType.FnFieldName): getNextPage}
+        func New@(Model.Name) (cur @(modelPageType.ContentType.Name), getNextPage @(modelPageType.NextPageFunctionSig)) @Model.Name {
+        return @(Model.Name){
+        @(modelPageType.FnFieldName): getNextPage,
+        @(modelPageType.ResultFieldName): cur,
+        }
```
Comment (ArcturusZhang) [inline src/Templates/ModelTemplate.cshtml:421]: Visual studio changes the format of the template every time I save the file... The only actual change in this file is here

Comment (jhendrixMSFT) [reply src/Templates/ModelTemplate.cshtml:421]: Yes the auto-formatting drives me nuts.

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
I assume you ran the tests locally and there were no issues?

**Comment by ArcturusZhang** [comment]
> I assume you ran the tests locally and there were no issues?

I tried locally generation for some packages and no compile error reported. Actually the constructor is not referenced within the whole SDK - therefore there should be no harm.

**Comment by fearthecowboy** [comment]
# 🤖 AutoRest automatic publish job 🤖
## success (version: 2.1.158)

**Pull request merged**